### PR TITLE
[Feature](inverted index) add tokenize function for inverted index

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

In this PR, we introduce TOKENIZE function for inverted index, it is used as following:
```
SELECT TOKENIZE('I love my country', 'english');
```
It has two arguments, first is text which has to be tokenized, the second is parser type which can be **english**, **chinese** or **unicode**.
It also can be used with existing table, like this:
```
mysql> SELECT TOKENIZE(c,"chinese") FROM chinese_analyzer_test;
+---------------------------------------+
| tokenize(`c`, 'chinese')              |
+---------------------------------------+
| ["来到", "北京", "清华大学"]          |
| ["我爱你", "中国"]                    |
| ["人民", "得到", "更", "实惠"]        |
+---------------------------------------+
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 53.81 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17162606603 Bytes

**Comment by airborne12** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 51.57 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17162027650 Bytes

File: be/src/vec/functions/function_tokenize.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    ColumnString::Chars& column_string_chars = dest_column_string.get_chars();
+    ColumnString::Offsets& column_string_offsets = dest_column_string.get_offsets();
+    column_string_chars.reserve(0);
+
+    ColumnArray::Offset64 string_pos = 0;
+    ColumnArray::Offset64 dest_pos = 0;
+    ColumnArray::Offset64 src_offsets_size = src_column_string.get_offsets().size();
+
+    InvertedIndexCtx inverted_index_ctx;
+    if (tokenize_type.to_string() == "chinese") {
```
Comment (xiaokang) [inline be/src/vec/functions/function_tokenize.cpp:55]: we can use util function to get ParserType from name.

**Review by xiaokang — COMMENTED** [review]
add regression testcase

**Comment by airborne12** [comment]
run buildall

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by airborne12** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.2 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17169720430 Bytes

**Review by qidaye — APPROVED** [review]
LGTM

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Pull request merged**